### ISSUE TITLE
pilot fails on Titan  

### ISSUE THREAD

**Issue by jdakka** [radical-cybertools/radical.pilot#1560]
An HTBAC script was submitted using `ornl.titan_aprun`. This run was accidentally submitted with `devel` version of `RP` instead of `master`. The `saga` debug logs are showing `pilot failed` but no additional errors. In this experiment, 25 replicas were submitted for 13 systems which translates to 325 concurrent CUs, for a total of 1300 CUs. Each CU uses 16 cores, plus an additional node for the `Agent`. Only 13 units finished executing, at which the pilot failed. The only error I found was in the `bootstrap.log` for an `MPI` abort. 

Resubmitting now with a smaller workload (25 concurrent CUs @ 64 cores per CU) and using the `master` branch of `RP`. 

```
 python               : 2.7.9
  pythonpath           : /sw/xk6/xalt/0.7.5/site:/sw/xk6/xalt/0.7.5/libexec
  virtualenv           : /lustre/atlas2/chm126/scratch/farkaspall/radical-master

  radical.pilot        : 0.47.1
  radical.utils        : 0.47-v0.47-3-g18dab2b@master
  saga                 : 0.47-v0.46-34-g0f186f6@master
```

[client_files.tgz.zip](https://github.com/radical-cybertools/radical.pilot/files/1737973/client_files.tgz.zip)

**Comment by andre-merzky** [comment]
This looks like  a problem on the pilot side - can you please add the agent logs?  Thanks!

**Comment by jdakka** [comment]
Sorry I tried adding it before but just saw that it was larger than 10 MB. I believe you can access 
`/lustre/atlas2/chm126/scratch/farkaspall/radical.pilot.sandbox/rp.session.titan-ext1.farkaspall.017578.0000.tgz.zip` directly on Titan? Let me know if you cannot, since this is going through the INSPIRE project folder.

**Comment by kristofarkas** [comment]
It seems that runs that have large number of concurrent CUs fail. Not all units are submitted, for example in this case 40/125 CU get a unit.XXX folder created, before the pilot fails.

Logs for this particular run: the client folder contains the `debug.log` too.
[client.session.titan-ext3.farkaspall.017582.0002.tar.gz](https://github.com/radical-cybertools/radical.pilot/files/1740080/client.session.titan-ext3.farkaspall.017582.0002.tar.gz)
[rp.session.titan-ext3.farkaspall.017582.0002.tar.gz](https://www.dropbox.com/s/8o3ze2cdl7abx8c/rp.session.titan-ext3.farkaspall.017582.0002.tar.gz?dl=0)

**Comment by kristofarkas** [comment]
In the logs the error shows this:
```
Traceback (most recent call last):
  File "/lustre/atlas2/chm126/scratch/farkaspall/radical-master/lib/python2.7/site-packages/radical/entk/execman/task_manager.py", line 260, in unit_state_cb
    self._mq_connection.close()
  File "/lustre/atlas2/chm126/scratch/farkaspall/radical-master/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 688, in close
    self._flush_output(self._closed_result.is_ready)
  File "/lustre/atlas2/chm126/scratch/farkaspall/radical-master/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 455, in _flush_output
    self._impl.ioloop.poll()
  File "/lustre/atlas2/chm126/scratch/farkaspall/radical-master/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 245, in poll
    self._poller.poll()
  File "/lustre/atlas2/chm126/scratch/farkaspall/radical-master/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 923, in poll
    events = self._poll.poll(self._get_next_deadline())
AttributeError: 'NoneType' object has no attribute 'poll'

2018-02-20 07:16:15,705: radical.entk.task_manager: MainProcess                     : MainThread     : INFO    : Task manager process closed
2018-02-20 07:16:15,706: radical.entk.appmanager: MainProcess                     : MainThread     : INFO    : Restarting task manager process from AppManager
2018-02-20 07:16:15,706: radical.entk.task_manager: MainProcess                     : MainThread     : WARNING : tmgr process already running, but attempted to restart!
2018-02-20 07:16:15,706: radical.entk.appmanager: MainProcess                     : MainThread     : INFO    : Restarting heartbeat thread from AppManager
2018-02-20 07:16:15,706: radical.entk.task_manager: MainProcess                     : MainThread     : WARNING : Heartbeat thread already running, but attempted to restart!
2018-02-20 07:16:15,706: radical.entk.appmanager: MainProcess                     : MainThread     : INFO    : Terminating heartbeat thread from AppManager
2018-02-20 07:16:15,706: radical.entk.task_manager: MainProcess                     : MainThread     : INFO    : Hearbeat thread terminated
2018-02-20 07:16:15,720: radical.entk.appmanager: MainProcess                     : MainThread     : INFO    : Restarting task manager process from AppManager
2018-02-20 07:16:15,720: radical.entk.task_manager: MainProcess                     : MainThread     : WARNING : tmgr process already running, but attempted to restart!
2018-02-20 07:16:15,720: radical.entk.appmanager: MainProcess                     : MainThread     : INFO    : Restarting heartbeat thread from AppManager
2018-02-20 07:16:15,720: radical.entk.task_manager: MainProcess                     : MainThread     : WARNING : Heartbeat thread already running, but attempted to restart!
2018-02-20 07:16:15,720: radical.entk.appmanager: MainProcess                     : MainThread     : INFO    : Terminating tmgr process from AppManager
2018-02-20 07:16:15,720: radical.entk.task_manager: MainProcess                     : MainThread     : INFO    : Task manager process closed
2018-02-20 07:16:15,720: radical.entk.appmanager: MainProcess                     : MainThread     : INFO    : Restarting task manager process from AppManager
2018-02-20 07:16:15,721: radical.entk.task_manager: MainProcess                     : MainThread     : WARNING : tmgr process already running, but attempted to restart!
2018-02-20 07:16:15,721: radical.entk.appmanager: MainProcess                     : MainThread     : INFO    : Restarting heartbeat thread from AppManager
2018-02-20 07:16:15,721: radical.entk.task_manager: MainProcess                     : MainThread     : WARNING : Heartbeat thread already running, but attempted to restart!
```

**Comment by andre-merzky** [comment]
Oh,  I am afraid that points to the EnTK layer - or rather to RabbitMQ.  Pinging @vivek-bala

**Comment by jdakka** [comment]
Creating new ticket in `EnTK` [Issue 21](https://github.com/radical-cybertools/radical.entk/issues/214)

**Comment by vivek-bala** [comment]
I got the same setup as jumana and kristoff but I am blocked by https://github.com/radical-cybertools/radical.pilot/issues/1566

**Comment by andre-merzky** [comment]
Vivek, #1566 is closed by now - will you take this issue up again?  Anything I can help with?

**Comment by jdakka** [comment]
@andre-merzky @vivek-bala 
I ran the same workload (as these issues are also halting my progress as well). It appears that the issue with Agent_1 not finding the VE is still persistent. Not sure if this conflates with any progressions on `EnTK`. Not sure if my testing right now helps in any sense either but just in case...

Ran with the latest of:

```
radical.pilot        : 0.47-0.47@master
radical.utils        : 0.47.1-v0.47.1@master
saga                 : 0.47-v0.46-34-g0f186f66@master
```
[rp.session.titan-ext5.jdakka.017588.0000_real_client.tgz.zip](https://github.com/radical-cybertools/radical.pilot/files/1756939/rp.session.titan-ext5.jdakka.017588.0000_real_client.tgz.zip)
[rp.session.titan-ext5.jdakka.017588.0000_agent.tgz.zip](https://github.com/radical-cybertools/radical.pilot/files/1756940/rp.session.titan-ext5.jdakka.017588.0000_agent.tgz.zip)
[records.tgz.zip](https://github.com/radical-cybertools/radical.pilot/files/1756942/records.tgz.zip)

**Comment by andre-merzky** [comment]
This was caused a bashrc entry which changed `PWD`  during bootstrapper startup - mark as resolved
